dev-java/parsington/parsington-3.1.0.ebuild

```diff
@@ -0,0 +1,46 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+JAVA_PKG_IUSE="doc source test"
+JAVA_TESTING_FRAMEWORKS="junit-jupiter"
+
+inherit java-pkg-2 java-pkg-simple
+
+DESCRIPTION="A mathematical expression parser for infix expression strings"
+HOMEPAGE="https://github.com/scijava/parsington"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/scijava/parsington.git"
+	S="${WORKDIR}/${P}"
+	MAVEN_ID="org.scijava:parsington:9999"
+else
+	SRC_URI="
+		https://github.com/scijava/${PN}/archive/refs/tags/${P}.tar.gz -> ${P}-sources.tar.gz
+	"
+	S="${WORKDIR}/${PN}-${P}"
+	MAVEN_ID="org.scijava:parsington:3.1.0"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="BSD-2"
+SLOT="0"
+
+BDEPEND="
+	>=virtual/jdk-1.8:*
+	test? (
+	      >=dev-java/junit-5.9.1
```

Comment:
Now it compiles fine, but there is still a warning:
```
* java-pkg_ensure-dep: The ebuild is attempting to use junit-5, which is not declared with a SLOT in DEPEND.
* java-pkg_ensure-dep: The ebuild is attempting to use junit-5, which is not declared with a SLOT in DEPEND.
```

We need slot dependency `:5` on the `junit` dependency. The warning seems to imply this should be in `DEPEND` instead of `BDEPEND` so maybe I was wrong earlier, or maybe the warning message will go away as well if the slot operator is added to `BDEPEND`.

Reply:
I just pushed https://github.com/gentoo/sci/pull/1200/commits/ad997359ec931fca721596d50c53df04ffba0f4b
I think that might suppress the warning by indicating any slot

regards

Reply:
`The warning seems to imply this should be in `DEPEND` instead of `BDEPEND` so maybe I was wrong earlier, or maybe the warning message will go away as well if the slot operator is added to `BDEPEND`.
`
I think BDEPEND is right, since is a test? dependency

Reply:
1. Is junit:5 (slot5 ) actually junit-5?
2. will the ebuild break against future junit-6?
Regards

Reply:
> 1. Is junit:5 (slot5 ) actually junit-5?

Yes, in this case it is

> will the ebuild break against future junit-6?

Yes, because we have `JAVA_TEST_GENTOO_CLASSPATH="junit-5"` in the ebuild.

Reply:
ok, I will send a fixed (and squashed) pull request then

Reply:
Hi @AndrewAmmerlaan 

Changed to

```
BDEPEND="
	>=virtual/jdk-1.8:*
	test? (
	      >=dev-java/junit-5.9.1:5
```

squashed the commit and updated the pull request,

Regards
@IAHM-COL

Reply:
@AndrewAmmerlaan 

Commit https://github.com/gentoo/sci/pull/1200/commits/b99c9c2e7aca7e09682e4da68f981284813fc934 might suppress the SLOT in DEPEND warning now


Regards
@IAHM-COL